homeassistant/components/sensor/qq.py

```diff
@@ -0,0 +1,102 @@
+"""You can use this sensor to link your qq."""
+import os
+import logging
+import threading
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['qqbot==2.3.7']
+
+_LOGGER = logging.getLogger(__name__)
+
+QQ_NUMBER = 'qq'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(QQ_NUMBER): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Qqsensors."""
+    thread1 = QQ(config.getp[QQ_NUMBER])
+    thread1.start()
+    object_qq = Qqsensor(hass, QQ_NUMBER, thread1)
+    add_devices([object_qq])
+
+
+class Qqsensor(Entity):
+    """Representation of a Qqsensor."""
+
+    def __init__(self, hass, name, thread_qq):
+        """Initialize the sensor."""
+        self._state = 'NULL'
+        self._name = name
+        self._hass = hass
+        self._thread = thread_qq
+        self._hass.services.register(self._name, 'stop qq', self.stop)
+        self._hass.services.register(self._name, 'run qq', self.run)
+
+    @property
+    def should_poll(self):
+        """Need polling."""
+        return True
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the message of the sensor."""
+        return self._state
+
+    def update(self):
+        """Get message from file."""
```

Comment:
Is is possible to get state from qqbot library directly without reading it from a file?

Reply:
I've tried to understand some of the library code. I don't even get how the bot knows to write to '~/.homeassistant/msg.txt'. This *could* be assuming that the working directory is set to '~/.homeassistant', i.e., you executed hass from there. That would break in a lot of installations.

Reply:
I agree. So the best would be if the library could provide the data directly without reading it from a file

Reply:
Absolutely. If that is not possible, I think the best thing would be to use `os.mkfifo` to create a named pipe and make the library use that as "file". That would mean we don't support Windows though. (Do we support Windows?)